noir-projects/aztec-nr/aztec/src/note/note_getter.nr

```diff
@@ -118,10 +118,18 @@ fn constrain_get_notes_internal<Note, N, M, FILTER_ARGS>(
 ) -> [Option<Note>; MAX_NOTE_HASH_READ_REQUESTS_PER_CALL] where Note: NoteInterface<N, M> {
     let mut returned_notes = [Option::none(); MAX_NOTE_HASH_READ_REQUESTS_PER_CALL];
 
+    // The filter is applied first to avoid pushing note read requests for notes we're not interested in. Note that
+    // while the filter function can technically mutate the contents of the notes (as opposed to simply removing some),
+    // the private kernel will later validate that these note actually exist, so transformations would cause for that
+    // check to fail.
+    let filter_fn = options.filter;
+    let filter_args = options.filter_args;
+    let filtered_notes = filter_fn(opt_notes, filter_args);
```

Comment:
Is my understanding correct that the "main fix" in this PR is that we move the filter_fn from unconstrained `get_notes_internal` to here and that the expectation is that the specific constraints will be layed down in filter_fn (e.g. that some value is less than 5)? And these constraints were ignored until now because it was called in the unconstrained function?

Reply:
Did you read the pr description? 👀 
> The application of the filter was not constrained when reading notes, this PR changes that.